### DISCUSSION TITLE
How to configure the development environment

### DISCUSSION THREAD

**Discussion by denerFernandes** [medusajs/medusa#1339] [Q&A] [answered]
Hi guys, I loved Medusa, it's very promising.

I want to develop two features:

- Add support for different languages for products and collections (I know about external cms, but in my case it's not option)
- Support for creating pages (as a visual composer)

I can submit a pull request if it is interesting for the project.

I don't have much experience with Lerna, can you explain to me how I can configure the development environment?

**Comment by srindom** [comment, marked as answer]
Hey!

Welcome happy you like what you have seen so far - would love to help you get started :)

First read through the contribution guide here: https://github.com/medusajs/medusa/blob/master/CONTRIBUTING.md that covers the basics.

When you have the Medusa monorepo cloned and a test project initialized with `medusa new` you should do the following:

Install `medusa-dev-cli`:
```
yarn add medusa-dev-cli
medusa-dev -p [the path to the root of your monorepo]
```

### In the monorepo

Build your packages 
```
yarn build
```

### In your test project

Copy over all the build files to your local project
```
medusa-dev
```

What happens now is that instead of pulling files from the npm registry `medusa-dev` will pull files from your local Medusa monorepo. So in the future when you make a change in one of the Medusa packages you can follow the same sequence: `yarn build`, `medusa-dev` to get your test project to use your local version.

To make this a bit less manual I typically `cd` into the package I am working on and run `yarn watch` - then the package will build every time I make a change. And then in my test project, I just allow `medusa-dev` to run after having copied over all the stuff - then it will listen for new files to copy over when the build has processed. 

> Tip: Sometimes you may encounter that packages are not being updated correctly. Without going into the full details of why this happens, the fix is typically to run `medusa-dev --force-install` :)

Hope this helps! 

In regards to the features you are looking to develop, I would love to hear more about the multi-language support; this would definitely be something we would like to include. Maybe you could write a couple of paragraphs on how you would approach the problem, so we can agree on something before you start coding? :)

**Reply by gaurav1999** [reply]
Hey @srindom , I followed this approach,  but a little bit confused about how to test and work on individual packages.
I wanted to contribute to pacakges/medusa.

Did wrote some code, but not able to figure out how to build and run test over it using this approach.

Thanks!

cc @olivermrbl

**Reply by gaurav1999** [reply]
`Copy over all the build files to your local project` , would like to get a little more help over this instruction.

**Reply by bytrangle** [reply]
Hi @srindom , thank you for the detailed walk through. I notice that the [contributing file](https://github.com/medusajs/medusa/blob/master/CONTRIBUTING.md) doesn't mention installing and using`medusa-dev-cli`, and listen for changes to local Medusa monorepo from a test project. Those elements are critical for development. Should I add them to the Contributing file?

**Reply by bytrangle** [reply]
@gaurav1999 You may have figured it out by now, but **Copy over all the build files to your local project** means you're in the test project directory, then run `medusa-dev`. That's all you need to do. Your project will use the built files from locally installed Medusa monorepo, instead of the published Medusa package on npm, which is a stable release and doesn't include your new codes. Does that make sense?

**Comment by denerFernandes** [comment]
Thanks @srindom  for tutorial.

I'll try :D

About languages, i first thought of working similarly at currencies, basically define which languages are available in the store, as well as a default language, such as the possibility of defining languages for each region, just like in currencies.

Then I think about moving the translated fields to another table in the database that will be related to the languages.

Same for collections and products (name, descriptions, handle)

Before doing this I need to define the impact that this change will have, for example in the meilisearch plugin.

This is my initial plan, can you tell me if this is a good plan for Medusa?

Any other points I have to pay attention to?

**Comment by paulm17** [comment]
Has anyone been able to get the dev env running?  

I've gone through the docs https://docs.medusajs.com/development/overview

But they aren't really clear.  

Do I start at backend quick start or go straight to local development?

Via backend quickstart, I have the store running.  But how to dev?

Via local development, I have all the packages installed but medusa-dev just exits and yarn watch medusa does nothing meaningful except show a list and medusa develop gives me a babel error.

```
medusa develop

Start development server. Watches file and rebuilds when something changes

Options:
  --verbose                Turn on verbose output                                                        [boolean] [default: false]
  --no-color, --no-colors  Turn off the color in output                                                  [boolean] [default: false]
  --json                   Turn on the JSON logger                                                       [boolean] [default: false]
  -H, --host               Set host. Defaults to localhost                                          [string] [default: "localhost"]
  -p, --port               Set port. Defaults to 9000                                                    [string] [default: "9000"]
  -h, --help               Show help                                                                                      [boolean]
  -v, --version            Show the version of the Medusa CLI and the Medusa package in the current project               [boolean]
babel:
  src does not exist
Error: Command failed: "/Users/paul/development/src/github/medusa/node_modules/.bin/babel" src -d dist
```

The docs could be a lot better here.

Thanks!
